tests/test_framework/electrs.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import os
+
+from ephemeral_port_reserve import reserve
+from test_framework.utils import BitcoinBackend, TailableProc, ELECTRS_PATH
+
+
+class Electrs(BitcoinBackend):
+    def __init__(
+        self,
+        bitcoind_dir,
+        bitcoind_rpcport,
+        bitcoind_p2pport,
+        electrs_dir,
+        rpcport=None,
+    ):
+        TailableProc.__init__(self, electrs_dir, verbose=False)
+
+        if rpcport is None:
+            rpcport = reserve()
+
+        self.electrs_dir = electrs_dir
+        self.rpcport = rpcport
+
+        regtestdir = os.path.join(electrs_dir, "regtest")
+        if not os.path.exists(regtestdir):
+            os.makedirs(regtestdir)
+
+        self.cmd_line = [
+            ELECTRS_PATH,
+            "--conf",
+            "{}/electrs.toml".format(regtestdir),
+        ]
+        electrs_conf = {
+            "daemon_dir": bitcoind_dir,
+            "cookie_file": os.path.join(bitcoind_dir, "regtest", ".cookie"),
+            "daemon_rpc_addr": f"127.0.0.1:{bitcoind_rpcport}",
+            "daemon_p2p_addr": f"127.0.0.1:{bitcoind_p2pport}",
+            "db_dir": electrs_dir,
+            "network": "regtest",
+            "electrum_rpc_addr": f"127.0.0.1:{self.rpcport}",
+        }
+        self.conf_file = os.path.join(regtestdir, "electrs.toml")
+        with open(self.conf_file, "w") as f:
+            for k, v in electrs_conf.items():
+                f.write(f'{k} = "{v}"\n')
+
+    def start(self):
+        TailableProc.start(self)
+        logging.info("Electrs started")
+
+    def startup(self):
+        try:
+            self.start()
+        except Exception:
+            self.stop()
+            raise
+
+    def stop(self):
+        return TailableProc.stop(self)
+
```

Comment:
i think we don't need to override this method, as we implicitly inherit from `TailableProc`

Reply:
Now that the electrs logs are working in tests, perhaps we should wait for "electrs stopped" in the logs as part of this method.